pkgs/development/libraries/arrow-cpp/default.nix

```diff
@@ -67,25 +68,29 @@ in stdenv.mkDerivation rec {
   '';
 
   cmakeFlags = [
-    "-DARROW_BUILD_TESTS=ON"
+    "-DARROW_BUILD_TESTS=${if static then "OFF" else "ON"}"
     "-DARROW_DEPENDENCY_SOURCE=SYSTEM"
+    "-DARROW_JEMALLOC=${if static then "OFF" else "ON"}"
     "-DARROW_PARQUET=ON"
     "-DARROW_PLASMA=ON"
-    "-DARROW_PYTHON=ON"
+    "-DARROW_PYTHON=${if static then "OFF" else "ON"}"
```

Comment:
For each case where you disable a feature on `static`, please add a comment on why the feature is disabled (e.g. because upstream doesn't support that feature in combination with static builds, or because the dependencies aren't packaged accordingly in nixpkgs, etc.).

This is so that this knowledge you gained on the way to this is not lost to the next person who may want to work on enabling that feature in conjunction with static builds.